core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -797,6 +798,7 @@ func testSetup(
 	lggr := logger.TestLogger(t)
 	lggr.SetLogLevel(zapcore.ErrorLevel)
 	db := pgtest.NewSqlxDB(t)
+	t.Cleanup(func() { assert.NoError(t, db.Close()) })
```

Comment:
It is important to register defer/Cleanup calls immeadiately, no only so they are not forgotten (like `lp2` was in this case), but because they must run after this point no matter how or when the test exits - which could be before the old Cleanup call at the end.

Reply:
cc @asoliman92